pgcli/main.py

```diff
@@ -440,11 +461,22 @@ def connect(self, database='', host='', user='', port='', passwd='',
             try:
                 pgexecute = PGExecute(database, user, passwd, host, port, dsn,
                                       application_name='pgcli', **kwargs)
-                if passwd:
+                if passwd and keyring and self.config["main"].get("keyring", True):
                     try:
                         keyring.set_password('pgcli', key, passwd)
-                    except (keyring.errors.InitError, RuntimeError):
-                        pass
+                    except (
```

Comment:
This keyring-related handling is getting long enough to be extracted into methods, or perhaps a class that `main` can delegate things to. Our `main` is very clattered and it's becoming difficult to follow the logic.

Reply:
Hmm yes, but I don't like to generalize a generalization (which I think keyring is). You're right about it's getting long and `main.py` is flooded with exceptions like this. But I don't know how that can be fixed.

I've moved `keyring.set_password` so it only has to be called once.